### ISSUE TITLE
Generated Code contains require -> no warnings

### ISSUE THREAD

**Issue by yanosz** [babel/babel#6807]
Current situation:
-----------------------------------
Transpiling  
```
import hello from './helloworld' 
```
Generates:
```
var _hello = require("./hello");
var _hello2 = _interopRequireDefault(_hello);
```
Without any warning. 

Problem
--------------- 
This is confusing, since the homepage says that:
"Put in next-gen JavaScript - Get browser-compatible JavaScript out" - however, hardly any browser supports `require(.)` out of the box.

Propsals
------------------------

1. (Preferred): Do not generate 'require (.)' by default. Implement namespaces for modules on a global module, only. Show a note, that all `.js` files have to be loaded by the HTML.
2. (Alternative): Keep `require`. Output a warning, that additional libraries such as require.js are needed. Provide a hit (HTML-Source-Code) how to include these.
3. (last resort): Output a warning, that the result is not "browser-compatible JavaScript", due to "require".

**Comment by hzoo** [comment]
We assume if you are using babel in the browser you are using a bundler like rollup/webpack, it's always been this way. Would be cool to add a warning but we would somehow have to know if you don't have a bundler, which I don't know how you would do atm. If you are building babel in the browser you need to use it with `babel-loader`, otherwise it's for node

**Comment by yanosz** [comment]
IMHO this is basically a usability issue and not about whether something like webpack is reasonable or not.

If we go for 2) imho it's practical to output a warning while having a command line switch for disabling specific warnings.

By that the default is helpful for users calling babel-cli manually, while 3rd party devlopers can disable this warning if needed.

**Comment by nicolo-ribaudo** [comment]
If you don't specify a module format and thus Babel output require() calls, it will crash as soon as you open your code in a browsers.
I don't think a warning is needed, since the configuration problem will be noticed in the very early stages of development.

**Comment by yanosz** [comment]
Well, I strongly disagree. If babel produces code that crashes, a warning certainly reasonable.

Warnings are supposed to help users avoiding long bug hunts.

Getting back to: "Put in next-gen JavaScript - Get browser-compatible JavaScript out" users expect to get browser-compatible JavaScript out of babel - by that, they're looking at ways on how to configure babel.

**Comment by xtuc** [comment]
I agree with your point. This is the first time I've heard a complain about it, I assume that most user know that bundling is required in that case.

We cannot really emit a warning a think because the bundling is usually a separate process.

**Comment by yanosz** [comment]
Providing some detauls:

Is:
```
es2015-demo/babel$ ./node_modules/.bin/babel -d build/ src
src/app.js -> build/app.js                                                                                                                              
src/hello.js -> build/hello.js
```

More helpful:
```
es2015-demo/babel$ ./node_modules/.bin/babel -d build/ src
src/app.js -> build/app.js
WARNING: Transpiling output contains require(.) - consider postprocessing using packers <LINK>                                                                                                                    
src/hello.js -> build/hello.js

es2015-demo/babel$ ./node_modules/.bin/babel --distable-warinings:require -d build/ src
src/app.js -> build/app.js
src/hello.js -> build/hello.js
```
Technically this is possible, IMHO.

**Comment by c-johnson** [comment]
Hi, I'm a basic dummy JS developer trying to "Babelify" my code (using gulp-babel) and am also confused by this issue. I thought that Babel would take my ES6 source code and convert it into usable Javascript, which it does not. So +1 for including this as a warning or changing the marketing for Babel to say "Put in next-gen JavaScript - Get bytecode for eventual browser-compatible JavaScript out with a bundler like Webpack or Rollup"

**Comment by nicolo-ribaudo** [comment]
You can output amd modules instead of commonjs. Then you only need to add an and loader (like RequireJS) to your HTML, without needing a bundler.
